attest/untrusted/src/lib.rs

```diff
@@ -55,12 +59,18 @@ impl QuotingEnclave {
     }
 
     pub fn target_info() -> Result<(TargetInfo, EpidGroupId), TargetInfoError> {
-        let mut qe_info = TargetInfo::default();
-        let mut gid = EpidGroupId::default();
-        match unsafe { sgx_init_quote(qe_info.as_mut(), gid.as_mut()) } {
-            sgx_status_t::SGX_SUCCESS => Ok((qe_info, gid)),
-            sgx_status_t::SGX_ERROR_BUSY => Err(TargetInfoError::QeBusy),
-            other_status => Err(TargetInfoError::Sgx(other_status.into())),
+        let gid = Self::epid_group_id()?;
+        #[cfg(feature = "sgx-sim")]
+        {
+            // The Intel QE and PCE provided with `libsgx-dcap-ql` only work on SGX
+            // hardware. For EPID there is a simulator implementation of
+            // [sgx_init_quote()](https://github.com/intel/linux-sgx/blob/1efe23c20e37f868498f8287921eedfbcecdc216/sdk/simulation/uae_service_sim/quoting_sim.cpp#L138)
+            // Unfortunately there doesn't seem to be a DCAP equivalent.
+            Ok((TargetInfo::default(), gid))
+        }
+        #[cfg(not(feature = "sgx-sim"))]
+        {
+            Ok((TargetInfo::for_quoting_enclave()?, gid))
```

Comment:
For this I think the logic belongs longer term down in `mc-sgx-dcap-ql`. However the way that the `sgx-sim` feature is enabled here in the `MobileCoin` repo it can't propagate to dependency repos:
```mermaid
sequenceDiagram
    cargo build->>+mc-attest-untrusted: Build crate With features
    mc-attest-untrusted->>mc-attest-untrusted: Resolve features
    mc-attest-untrusted->>+mc-sgx-dcap-ql: Build crate With features
    mc-sgx-dcap-ql->>-mc-attest-untrusted: Propagate `CARGO_DEP` values
    mc-attest-untrusted->>mc-attest-untrusted: Invoke build.rs
    Note over mc-attest-untrusted: build.rs adds rustc-cfg=feature="sgx-sim"
    mc-attest-untrusted->>-mc-attest-untrusted: Compile with rustc
```
In the above diagram one can see that the `mc-attest-untrusted` crate doesn't enable the `sgx-sim` feature until it's `build.rs` is run, which is after `mc-sgx-dcap-ql` is built.

I punted for now on trying to hook these 2 up nicer. We can either try and key off the same environment variable as the MobileCoin repo or we could look at a more explicit enabling of the `sgx-sim` feature via a cargo build command line or an alias with `.cargo/config.toml`

Reply:
Personally I think the env var solution is fine, explicitly specifying features in cargo build commands feels more annoying to me since you have to keep remembering to do it, and I'm not sure how you would selectively enable it in `.cargo/config.toml`

Reply:
Sounds good i'll look to making the sgx n'sync with the main repo